sdk/core/azure-core/azure/core/settings.py

```diff
@@ -102,14 +114,52 @@ def convert_logging(value):
     val = value.upper()  # type: ignore
     level = _levels.get(val)
     if not level:
-        raise ValueError(
-            "Cannot convert {} to log level, valid values are: {}".format(
-                value, ", ".join(_levels)
-            )
-        )
+        raise ValueError("Cannot convert {} to log level, valid values are: {}".format(value, ", ".join(_levels)))
     return level
 
 
+def get_opencensus_span():
+    # type: () -> OpenCensusSpan
+    """Returns the OpenCensusSpan if opencensus is installed else returns None"""
+    try:
+        from azure.core.tracing.ext.opencensus_span import OpenCensusSpan
+
+        return OpenCensusSpan
+    except ImportError:
+        return None
+
+
+def get_opencensus_span_if_opencensus_is_imported():
+    if "opencensus" not in sys.modules:
+        return None
+    return get_opencensus_span()
+
+
+_tracing_implementation_dict = {"opencensus": get_opencensus_span()}
+
+
+def convert_tracing_impl(value):
+    # type: (Union[str, AbstractSpan]) -> AbstractSpan
+    """Convert a string to AbstractSpan
+
+    If a AbstractSpan is passed in, it is returned as-is. Otherwise the function
+    understands the following strings, ignoring case:
+
+    * "opencensus"
+
+    :param value: the value to convert
+    :type value: string
+    :returns: AbstractSpan
+    :raises ValueError: If conversion to AbstractSpan fails
+
+    """
+    if value is None:
+        return get_opencensus_span_if_opencensus_is_imported()
+
+    value = value.lower() if isinstance(value, six.string_types) else value
+    return _tracing_implementation_dict.get(value, value)
```

Comment:
What's the rationale for returning the key as a default value? Offhand if I pass an impl name that does not exist or is not installed I would expect either an error, or alternatively a warning and it would act as is no impl was specified, not to get the impl name back

Reply:
The idea is that the user can also pass in their own wrapper plugin as they desire. If we had gone with an abstract class, then I would be able to check if what they pass in was a subclass. But since we used a protocol for the abstract span instead, we must just trust that the user will just pass in the correct values.

I made a change that will do at least some error handling. Let me know if you find it disagreeable.

Reply:
All the dictionary keys are strings? We could just check for "not string" and then assume that whatever was passed in is a custom wrapper if it is not a string

Reply:
yes that is what I have just changed to and just pushed.